### PR TITLE
Delegations frontoffice prototype

### PR DESCRIPTION
**Pull request by sauloperez**
#### :tophat: What? Why?

This is a very early WIP of the proposal we discussed with CoopCat last week. **Disclaimer: it will change**. At this point, we want to know which UX provides the best balance between usability for users and maintainability cost.

This relies on https://github.com/coopdevs/decidim-module-action_delegator/pull/15.

### :camera: Screenshots (optional)
So far this looks like:

Peek 2020-09-10 11-37

### CODE REVIEW

File: decidim-consultations/app/assets/javascripts/decidim/consultations/vote_dialog.js [hunk 1]
```diff
@@ -46,4 +46,18 @@ $(document).ready(function () {
...
+      delegationField = $("#decidim_consultations_delegation_id");
+
+  delegationsButton.click(function () {
+    delegationDialog.foundation("open");
+  });
+
+  delegationVoteButtons.click(function () {
+    voteDialog.foundation("open");
+    delegationField.val($(this).data("delegation-id"));
+  });
```
Comment (sauloperez) [inline decidim-consultations/app/assets/javascripts/decidim/consultations/vote_dialog.js]: this needs to be extracted into a JS file in https://github.com/coopdevs/decidim-module-action_delegator/.

**Comment by microstudi** [comment]
Another possibility is to add a step after changing to a delegated user.
Instead of showing the vote popup directly, reload the votation page and show in there clearly that you are going to vote for the delegated user.

**Comment by sauloperez** [comment]
I was thinking of doing that within the vote modal itself. I'll try to do that next.

**Comment by sauloperez** [comment]
Peek 2020-09-10 19-49

What do you think @microstudi ? I think it's pretty clear this way. I'm just worried about yet another "extension point" we need from Decidim to render our partial or copy the views and maintain them in our module (I don't know what's worse) :see_no_evil:

**Comment by microstudi** [comment]
Not bad @sauloperez I think we can go with that. We will see if after the tests this needs to be changed in any case!

**Comment by sauloperez** [comment]
I'm quite happy with the solution we have found as I'm not encountering edge cases and delegations seem to be well isolated from the rest of the UI. However, I'll ask you to discuss together what the next steps should be.

**Comment by sauloperez** [comment]
A little update on how it's looking so far 
Peek 2020-09-15 11-01

File: decidim-consultations/app/controllers/decidim/consultations/question_votes_controller.rb [hunk 2]
```diff
@@ -29,14 +29,40 @@ def create
...
+      private
+
+      def delegation
+        @delegation ||= Decidim::ActionDelegator::Delegation.find_by(id: params[:decidim_consultations_delegation_id])
+      end
+
+      def enforce_permission_to_unvote(subject, extra_context = {})
+        if delegation.blank?
+          enforce_permission_to :unvote, :question, question: current_question
+        else
```
Comment (sauloperez) [inline decidim-consultations/app/controllers/decidim/consultations/question_votes_controller.rb:50]: this else branch is the only bit that we needed to add.

**Comment by sauloperez** [comment]
We want with a different approach in https://github.com/coopdevs/decidim-module-action_delegator/pull/18 that makes it unnecessary to maintain a fork (thank God :joy: ). We'll keep developing these features in https://github.com/coopdevs/decidim-module-action_delegator/pull/15.
